File: C++/프로그래머스/0/120825. 문자 반복 출력하기/문자 반복 출력하기.cpp
#include <string>

using namespace std;

string solution(string my_string, int n) {
    string answer = "";
    
    // 미리 공간을 할당 (최적화)
    answer.reserve(my_string.length() * n);
    
    for(const auto c: my_string) {
        answer.append(n, c);
    }
    
    return answer;
}

File: Go/프로그래머스/0/120825. 문자 반복 출력하기/문자 반복 출력하기.go
func solution(my_string string, n int) string {
    res := make([]rune, 0, len(my_string) * n)
    
    for _, v := range my_string {
        for i := 0; i < n; i++ {
            res = append(res, v)
        }
    }
    
    return string(res)
}